roles/jd-client/src/job_declarator/mod.rs

```diff
@@ -201,15 +203,22 @@ impl JobDeclarator {
         self_mutex: &Arc<Mutex<Self>>,
         template: NewTemplate<'static>,
         token: Vec<u8>,
-        tx_list: Seq064K<'static, B016M<'static>>,
+        tx_list_sequence: Seq064K<'static, B016M<'static>>,
         excess_data: B064K<'static>,
-        coinbase_pool_outpust: Vec<u8>,
+        coinbase_pool_output: Vec<u8>,
     ) {
         let (id, _, sender) = self_mutex
             .safe_lock(|s| (s.req_ids.next(), s.min_extranonce_size, s.sender.clone()))
             .unwrap();
         // TODO: create right nonce
         let tx_short_hash_nonce = 0;
+        let mut tx_list: Vec<Transaction> = Vec::new();
+        for tx in tx_list_sequence.to_vec() {
+            //TODO remove unwrap
+            //TODO check if we want deserialize or consensus_decode
+            let tx = Transaction::deserialize(&tx).unwrap();
```

Comment:
@Fi3 it is fine here to use deserialize? should I use consensus_decode?

Reply:
deserialize call consensus_decode_from_finite_reader so is ok

Reply:
add an issue to add logic that manage unhappy path instead of unwrapping.